### PR TITLE
#1347: Remove returnFormat prop from DatePicker (closes #1347)

### LINKED ISSUE #1347: Remove returnFormat prop from DatePicker
@gabro is proposing to remove `returnFormat` (and `T`s other than `Date`) completely

_Originally posted by @giogonzo in https://github.com/buildo/react-components/pull/1333_

### PR DESCRIPTION
**Pull request by veej**
Closes #1347

### tests performed

Example in showroom still works as expected

### CODE REVIEW

**Comment by giogonzo** [comment]
my bad, the comment in https://github.com/buildo/react-components/pull/1333#discussion_r280793367 wasn't really clear. I think @gabro already has a PR in the making including this change?

**Comment by gabro** [comment]
@giogonzo not yet, I'm still working on the DateRangePicker component but I haven't touched the DatePicker yet.

File: src/DatePicker/DatePicker.tsx [hunk 1]
```diff
@@ -6,50 +6,35 @@ import { props, t, ReactChild } from "../utils";
...
+    onChange?: (date?: Date) => void;
     /** called when datepicker is closed */
     onHide?: () => void;
     /** MomentJS format used to display current date */
     displayFormat?: string;
-    /** called when value is cleared */
-    onClear?: () => void;
     /** minimum date selectable by the user */
-    minDate?: Value;
+    minDate?: Date | string;
```
Comment (gabro) [inline src/DatePicker/DatePicker.tsx]: is there a reason to also keep `string` in input? Can't we just stick to `Date`

Comment (veej) [reply src/DatePicker/DatePicker.tsx]: no particular reason. I left it just for convenience: there's no complex logic around these inputs and you have to convert it to a `Moment` date anyway, so I decided to keep everything that can be passed to `moment()`.
btw I'm ok to remove them if you prefer.

Comment (gabro) [reply src/DatePicker/DatePicker.tsx]: I prefer removing it. Having a stricter interface will make our projects more consistent and it allows changing the internals more freely.

**Pull request merged**